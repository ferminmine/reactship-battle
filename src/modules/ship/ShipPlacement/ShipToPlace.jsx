import React from 'react';
import withStyles from 'react-jss';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { nextShipToPlaceSelector } from '../ShipSelectors';

const styles = {
    shipToPlaceContainer: {
      marginBottom: '2em',
      textAlign: 'center'
    },
    nextShip: {
      color: '#2f9ad4'
    },
    shipName: {
      color: '#7b7979',
      fontWeight: 300,
      marginTop: 0
    },
    shipImage: {
      height: '5em'
    }
}

class ShipToPlace extends React.Component {
    render = () => {
        const nextShip = this.props.nextShipToPlace && this.props.shipsToPlace[0];
        const { classes } = this.props;
        return (
          <div className={classes.shipToPlaceContainer}>
            <h1 className={classes.nextShip}> 
              { nextShip !== undefined ? 'Next Ship to Place' : 'No more ships available' } 
            </h1>
            { 
              nextShip !== undefined && ( 
              <React.Fragment>
                <img className={classes.shipImage} alt='' src={nextShip && nextShip.image} />
                <h2 className={classes.shipName}> {nextShip && nextShip.name} </h2>
              </React.Fragment>
              )}
          </div>
        )
    }
}

const mapStateToProps = state => ({
  shipsToPlace: state.ships.shipsToPlace,
  nextShipToPlace: nextShipToPlaceSelector(state)
})

export default compose(
  withStyles(styles),
  connect(mapStateToProps)
)(ShipToPlace);