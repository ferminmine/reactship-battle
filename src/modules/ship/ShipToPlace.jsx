import React from 'react';
import withStyles from 'react-jss';
import { mapStateToProps, connect } from 'react-redux'
import { compose } 'reddux';
import PropTypes from 'prop-types';

const styles = {
    shipToPlaceContainer: {
      marginBottom: '2em',
      textAlign: 'center'
    },
    nextShip: {
      color: '#2f9ad4'
    },
    shipName: {
      color: '#7b7979',
      fontWeight: 300
    }
};

const mapStateToProps = (state) => ({
  shipsToPlace: state.ships.shipsToPlace
});

class ShipToPlace extends React.Component {

    static propTypes = {
      shipsToPlace: PropTypes.
    }

    render = () => {
        const nextShip this.props.shipsToPlace[0]
        const { classes } = this.props;
        return (
          <div className={classes.shipToPlaceContainer}>
            <h1 className={classes.nextShip}> Next Ship to Plase </h1>
            <img src={nextShip.image} />
            <h2 className={classes.shipName}> { nextShip.name } </h2>
          </div>
        )
    }
}

export default compose(
  withStyles(styles),
  connect(mapStateToProps)
)(ShipToPlace);