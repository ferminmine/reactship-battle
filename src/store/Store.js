import { createStore, combineReducers, compose, applyMiddleware } from 'redux';
import { createBrowserHistory } from 'history';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import playerReducer from '../modules/player/PlayerReducer';
import shipReducer from '../modules/ship/ShipReducer';
import gameReducer from '../modules/game/GameReducer';
import shootsReducer from '../modules/shoots/ShootsReducer';

export const history = createBrowserHistory();

const allReducers = combineReducers({
  player: playerReducer,
  game: gameReducer,
  ships: shipReducer,
  shoots: shootsReducer,
  router: connectRouter(history)
});

export const ROOT_REDUCER_ACTION_TYPES = {
  RESET_APP: 'RESET_APP'
};
export const resetApp = () => ({ type: ROOT_REDUCER_ACTION_TYPES.RESET_APP });

const rootReducer = (state, action) => {
  if (action.type === 'RESET_APP') {
    state = undefined;
  }

  return allReducers(state, action);
};

const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default createStore(
  rootReducer,
  composeEnhancer(applyMiddleware(routerMiddleware(history)))
);
