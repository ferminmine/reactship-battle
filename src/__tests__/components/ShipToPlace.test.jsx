import React from 'react';
import { mount } from 'enzyme';
import ShipToPlace from '../../modules/ship/ShipToPlace';
import { Provider } from 'react-redux';
import store from '../../store/Store';
import renderer from 'react-test-renderer';

describe('ShipToPlace Component Test', () => {
  it('renders without crashing', () => {
    mount(
      <Provider store={store}>
        {' '}
        <ShipToPlace />{' '}
      </Provider>
    );
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ShipToPlace />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
