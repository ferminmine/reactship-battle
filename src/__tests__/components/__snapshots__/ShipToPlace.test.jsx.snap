// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShipToPlace Component Test matches the snapshot 1`] = `
<div
  className="Connect(ShipToPlace)-shipToPlaceContainer-0-1-1"
>
  <h1
    className="Connect(ShipToPlace)-nextShip-0-1-2"
  >
     Next Ship to Plase 
  </h1>
  <img
    src="/imgs/cruiser.png"
  />
  <h2
    className="Connect(ShipToPlace)-shipName-0-1-3"
  >
     
    Cruiser
     
  </h2>
</div>
`;
